week-3/introduction_to_java_book_tasks/_10_3_my_integer/src/main/java/job_hacker/tasks/models/MyInteger.java

```diff
@@ -0,0 +1,70 @@
+package job_hacker.tasks.models;
+
+public class MyInteger {
+    private int value;
+
+    public MyInteger(int value) {
+        this.value = value;
+    }
+    public int getValue() {
+        return value;
+    }
+    public boolean isEven(){
+        return (value & 1) == 0;
+    }
+    public boolean isOdd(){
+        return (value & 1) == 1;
+    }
+    public boolean isPrime(){
+        if (value <= 1) {
+            return false;
+        }
+        if (value % 2 == 0) {
+            return value == 2;
```

Comment:
return true not 2

Reply:
Why?
That will result in failure in many test cases.

Reply:
Ok, you are right